clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -0,0 +1,4561 @@
+/* **********************************************************
```

Comment:
Note that I tried a fancy sequence at
https://stackoverflow.com/questions/16937359/git-copy-file-preserving-history
to preserve history in the copy w/o requiring `--follow` but it seems to have too many downsides.  It looks like the github review view here doesn't have `--follow` as it seems to think this is all new code.

Reply:
Hard to do a close review with the large diff (tried some quick diff techniques). I assume it's mostly just moving code like the PR description says, and that the unit tests passing is a good enough signal of correctness.